RevenueCatUI/Purchasing/PurchaseHandler.swift

```diff
@@ -380,6 +386,15 @@ extension PurchaseHandler {
         return true
     }
 
+    @MainActor
+    func updatePaywallSource(_ source: PaywallSource?) {
+        self.paywallSource = source
+
+        if let source {
+            self.eventData?.source = source.rawValue
+        }
+    }
```

Comment:
I did this to avoid modifying `PaywallEvent`. Now I am thinking that maybe we should call this `offeringSource`, rather than `paywallSource`